src/axom/sidre/core/MFEMSidreDataCollection.cpp

```diff
@@ -1430,6 +1425,476 @@ mfem::Geometry::Type MFEMSidreDataCollection::getElementTypeFromName(
+       */
+      bool operator!=(const VectorSpanIterator& other)
+      {
+        return (m_ptr != other.m_ptr) || (m_stride != other.m_stride);
+      }
+
+      /**
+       * @brief Advances the iterator to the next vector in the list
+       */
+      void operator++()
+      {
+        // Moves the pointer forward by the stride
+        m_ptr += m_stride;
+      }
+
+      /**
+       * @brief Returns the pointer to the start of the current
+       * vector - accesses on this pointer are not bounds-checked
+       */
+      const int* operator*() const { return m_ptr; }
+
+    private:
+      const int* m_ptr;
+      const IndexType m_stride;
+    };
+
+    VectorSpanIterator begin() const { return {m_data, m_num_components}; }
+    VectorSpanIterator end() const
+    {
+      return {m_data + (m_num_vectors * m_num_components), m_num_components};
+    }
+
+  private:
+    // Use reasonable defaults to avoid extra logic for empty spans
+    const int* m_data = nullptr;
+    IndexType m_num_vectors = 0;
+    IndexType m_num_components = 0;
+  };
+
+  /**
+   * @brief A data class containing all relevant information needed on geometric
+   * communication groups within an mfem::ParMesh - the shared geometric
+   * entities and neighboring groups
+   */
+  struct SharedGeometries
+  {
+    VectorSpan shared_verts;  // Indices of shared vertices
+    VectorSpan shared_edges;  // Pairs of vertex indices corresponding to shared edges
+    VectorSpan shared_triangles;  // 3-tuples of vertex indices corresponding to shared triangular faces
+    VectorSpan shared_quadrilaterals;  // 4-tuples of vertex indices corresponding to shared quadrilateral faces
+    mfem::Array<int> neighbors;  // Rank IDs of the neighboring nodes/processes
+  };
+
+  /**
+   * @brief Initializes the mfem::GroupTopology object
+   * @param [in] shared_geoms The shared geometry information
+   * @pre The shared vertex and neighbor fields of each SharedGeometries must be populated
+   * @post The gtopo member of the base subobject is initialized
+   */
+  void InitializeGroupTopology(const std::vector<SharedGeometries>& shared_geoms)
+  {
+    mfem::ListOfIntegerSets group_integer_sets;
+
+    // The first group always contains only the current rank
+    mfem::IntegerSet first_set;
+    mfem::Array<int>& first_array = first_set;  // Bind a ref so we can modify it
+    first_array.Append(MyRank);
+    group_integer_sets.Insert(first_set);
+
+    for(const auto& shared_geom : shared_geoms)
+    {
+      mfem::IntegerSet integer_set;
+      mfem::Array<int>& array = integer_set;  // Bind a ref so we can modify it
+      // On a given rank, the group ID corresponding to the current rank
+      // is a member of each group
+      array.Append(MyRank);
+      array.Append(shared_geom.neighbors);
+      array.Sort();  // MFEM requires that the sets be sorted
+      group_integer_sets.Insert(integer_set);
+    }
+    // FIXME: 822 or 823? No clue what either refers to
```

Comment:
lol.. where did this number come from?

Reply:
I dug down a bit into the MFEM impl and it looks like just arbitrary MPI tags.